### PR TITLE
executors: doc improvements

### PR DESCRIPTION
**Pull request by kalanchan**
builds upon and fixes merge conflicts from https://github.com/sourcegraph/sourcegraph/pull/42746.

closes https://github.com/sourcegraph/sourcegraph/issues/41516

## Test plan
N/A - docs change

### CODE REVIEW

File: doc/admin/executors/deploy_executors_binary.md [hunk 1]
```diff
@@ -183,3 +183,39 @@ systemctl enable executor
...
+
+Upgrading executors is relatively uninvolved. Simply follow the instructions below.
+Also, check the [changelog](https://github.com/sourcegraph/sourcegraph/blob/main/CHANGELOG.md) for any Executors related breaking changes or new features that you might want to configure.
+
+### **Step 1:** First, grab the executor binary for the new target Sourcegraph version.
+
+> NOTE: Keep in mind that only one minor version bumps are guaranteed to be disruption-free.
+
+```bash
+curl -sfLo executor https://storage.googleapis.com/sourcegraph-artifacts/executor/latest/linux-amd64/executor
```
Comment (danieldides) [inline doc/admin/executors/deploy_executors_binary.md:197]:
```suggestion
curl -sfLo executor https://storage.googleapis.com/sourcegraph-artifacts/executor/${SOURCEGRAPH_VERSION}/linux-amd64/executor
```

`latest` may not be the desired versino

File: doc/admin/executors/deploy_executors_binary.md [hunk 2]
```diff
@@ -183,3 +183,39 @@ systemctl enable executor
...
+chmod +x executor
+# Assuming /usr/local/bin is in $PATH.
+mv executor /usr/local/bin
+```
+
+### **Step 2:** Make sure all ambient dependencies and configurations are up-to-date:
```
Comment (eseliger) [inline doc/admin/executors/deploy_executors_binary.md:203]: this step requires some env vars to be set already, so maybe we should mention that.

File: doc/admin/executors/deploy_executors_binary.md [hunk 3]
```diff
@@ -81,11 +81,11 @@ mv executor /usr/local/bin
...
 
 | Env var                                  | Description                                                                                                                                                                                                                        | Example value                              |
-|------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------|
+| ---------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------ |
```
Comment (Piszmog) [inline doc/admin/executors/deploy_executors_binary.md:84]: The linter is going it get mad about this change. It doesn't like spaces in the column formatting.

Same goes for all the tables that got updated to have spaces

File: doc/admin/executors/deploy_executors_binary.md [hunk 4]
```diff
@@ -81,11 +81,11 @@ mv executor /usr/local/bin
...
 | `EXECUTOR_QUEUE_NAME`                    | The name of the queue to pull jobs from to. Possible values: `batches` and `codeintel` **required**                                                                                                                                | `batches`                                  |
-| `EXECUTOR_USE_FIRECRACKER`               | Whether to isolate jobs in virtual machines. Requires ignite and firecracker. Linux hosts only. Kubernetes is not supported. (default value: "true" when OS is Linux and not on Kubernetes)                                        | `true`                                     |
+| `EXECUTOR_USE_FIRECRACKER`               | Whether to isolate jobs in virtual machines. Requires ignite and firecracker. Linux hosts only. (default value: "true")                                                                                                            | `true`                                     |
```
Comment (Piszmog) [inline doc/admin/executors/deploy_executors_binary.md:88]: Should this be reverted - should we have this disclaimer about K8s not supporting Firecracker?

File: doc/admin/executors/deploy_executors_binary.md [hunk 5]
```diff
@@ -183,3 +183,39 @@ systemctl enable executor
 ### **Step 6:** Start receiving workloads
 
 If you use the systemd service, simply run `systemctl start executor`, otherwise run `executor run`. Your executor should start listening for jobs now! All done!
+
+## Upgrading executors
+
+Upgrading executors is relatively uninvolved. Simply follow the instructions below.
+Also, check the [changelog](https://github.com/sourcegraph/sourcegraph/blob/main/CHANGELOG.md) for any Executors related breaking changes or new features that you might want to configure.
```
Comment (Piszmog) [inline doc/admin/executors/deploy_executors_binary.md:190]: Should we use https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/CHANGELOG.md as the link to the changelog?

**Review by Piszmog — COMMENTED** [review]
Looks great! Just a couple questions

**Review by Piszmog — APPROVED** [review]
Nice job! 🔥

**Review by danieldides — APPROVED** [review]

**Comment by kalanchan** [comment]
IM BACK IN THE GAME BOYS

**Pull request merged**